task/06-conditions-n-loops-tasks.js

```diff
@@ -373,7 +461,26 @@ function toNaryString(num, n) {
  *   ['/web/favicon.ico', '/web-scripts/dump', '/webalizer/logs'] => '/'
  */
 function getCommonDirectoryPath(pathes) {
-    throw new Error('Not implemented');
+    let str = '';
+    for (let i = 0; i < pathes[0].length; ++i) {
+        str += pathes[0][i];
+        let c = 0;
+        for (let j=0; j < pathes.length; ++j) {
+            if (pathes[j].indexOf(str)!=0) {
+                c = 1;
+            }
+        }  
+        if (c == 1) {
+            break; 
+        } 
+    }
+    str = str.slice(0, -1);
+    while (str.length != 0 && str[str.length - 1] != '/') {
```

Comment:
you already know the index of the last common charater for paths. Just use this index to get a string from an orignal with single `slice `call
![image](https://user-images.githubusercontent.com/7272208/87436519-d0dc9d80-c5f5-11ea-84fe-cd3038fa769c.png)

Reply:
I changed my pathes task solution a bit because previous solution is a little confusing and adaptable only to task tests. I hope my code is more flexible now